core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -185,7 +254,30 @@ export class OptimizedSsrEngine {
             // store the render for future use
             this.renderingCache.store(renderingKey, err, html);
           }
-        });
+
+          if (
+            this.ssrOptions?.reuseCurrentRendering &&
+            !waitingRendersProcessing
+          ) {
+            this.log(
+              `Processing waiting SSR requests for ${request.originalUrl}...`
+            );
+            this.waitingRenderCallbacks[renderingKey]?.forEach((cb) =>
+              cb(err, html, true)
+            );
+            this.waitingRenderCallbacks[renderingKey] = null;
+          }
+        };
+
+        if (
+          this.ssrOptions?.reuseCurrentRendering &&
+          this.waitingRenderCallbacks[renderingKey]
+        ) {
+          this.waitingRenderCallbacks[renderingKey]?.push(renderCallback);
+        } else {
+          this.waitingRenderCallbacks[renderingKey] = [];
+          this.expressEngine(filePath, options, renderCallback);
+        }
       } else {
         // if there is already rendering in progress, return the fallback
```

Comment:
nitpick: this comment is no longer valid here. can be removed.
It would be nice to explain in JSDoc the `shouldRender`method, something for example: _when returns true, the server side rendering should be performed. When returns false, the CSR fallback should be returned._

Reply:
Good catch. 
Removed this comment, and added jsdoc for `shouldRender`